docs/rst/tutorials/micro/custom_platforms/custom_platforms.rst

```diff
@@ -0,0 +1,341 @@
+
+
+2. Build this environment locally using ``colcon build --cmake-args -DBUILD_TESTING=OFF``
+3. Source the local environment using ``source install/local_setup.bash``
+4. Create a new micro-ROS workspace and clone the micro-ROS Client packages inside it
+
+  - ``micro-CDR`` (`https://github.com/eProsima/micro-CDR <https://github.com/eProsima/micro-CDR>`_)
+  - ``Micro-XRCE-DDS-Client`` (`https://github.com/eProsima/Micro-XRCE-DDS-Client <https://github.com/eProsima/Micro-XRCE-DDS-Client>`_)
+  - ``rcl`` (`https://github.com/micro-ROS/rcl <https://github.com/micro-ROS/rcl>`_)
+  - ``rclc`` (`https://github.com/ros2/rclc <https://github.com/ros2/rclc>`_)
+  - ``rcutils`` (`https://github.com/micro-ROS/rcutils <https://github.com/micro-ROS/rcutils>`_)
+  - ``micro_ros_msgs`` (`https://github.com/micro-ROS/micro_ros_msgs <https://github.com/micro-ROS/micro_ros_msgs>`_)
+  - ``rmw-microxrcedds`` (`https://github.com/micro-ROS/rmw-microxrcedds <https://github.com/micro-ROS/rmw-microxrcedds>`_)
+  - ``rosidl_typesupport`` (`https://github.com/micro-ROS/rosidl_typesupport <https://github.com/micro-ROS/rosidl_typesupport>`_)
+  - ``rosidl_typesupport_microxrcedds`` (`https://github.com/micro-ROS/rosidl_typesupport_microxrcedds <https://github.com/micro-ROS/rosidl_typesupport_microxrcedds>`_)
+  - ``rosidl`` (`https://github.com/ros2/rosidl <https://github.com/ros2/rosidl>`_)
+  - ``rmw`` (`https://github.com/ros2/rmw <https://github.com/ros2/rmw>`_)
+  - ``rcl_interfaces`` (`https://github.com/ros2/rcl_interfaces <https://github.com/ros2/rcl_interfaces>`_)
+  - ``rosidl_defaults`` (`https://github.com/ros2/rosidl_defaults <https://github.com/ros2/rosidl_defaults>`_)
+  - ``unique_identifier_msgs`` (`https://github.com/ros2/unique_identifier_msgs <https://github.com/ros2/unique_identifier_msgs>`_)
+  - ``common_interfaces`` (`https://github.com/ros2/common_interfaces <https://github.com/ros2/common_interfaces>`_)
+  - ``test_interface_files`` (`https://github.com/ros2/test_interface_files <https://github.com/ros2/test_interface_files>`_)
+  - ``rmw_implementation`` (`https://github.com/ros2/rmw_implementation <https://github.com/ros2/rmw_implementation>`_)
+  - ``rcl_logging`` (`https://github.com/ros2/rcl_logging <https://github.com/ros2/rcl_logging>`_)
+  - ``ros2_tracing`` (`https://gitlab.com/ros_tracing/ros2_tracing <https://gitlab.com/ros_tracing/ros2_tracing>`_)
+  - ``micro_ros_utilities`` (`https://github.com/micro-ROS/micro_ros_utilities <https://github.com/micro-ROS/micro_ros_utilities>`_)
+
+5. Make sure that the following packages are removed or ignored:
+
+  - ``rosidl/rosidl_typesupport_introspection_cpp``
+  - ``rcl_logging/rcl_logging_log4cxx``
+  - ``rcl_logging/rcl_logging_spdlog``
+  - ``rclc/rclc_examples``
+  - ``rcl/rcl_yaml_param_parser``
+
+6. Build the micro-ROS workspace using colcon, your required ``colcon.meta`` and your custom compiler flags (using a CMake toolchain):
```

Comment:
Add a explanation, comment or reference that `my_toolchain.cmake` is the toolchain with the compiler and flags definitions.

Reply:
7c1157e55104330f4684d2315d6b33555353911d